tool_path_planner/src/utilities.cpp

```diff
@@ -0,0 +1,206 @@
+      const geometry_msgs::Point& point){
+    pcl::PointXYZ p;
+    std::tie(p.x,p.y,p.z) = std::make_tuple(point.x,point.y,point.z);
+    return p;
+  });
+
+  sensor_msgs::PointCloud2 cloud_msg;
+  pcl::toROSMsg(mesh_points,cloud_msg);
+  pcl_conversions::toPCL(cloud_msg,mesh.cloud);
+  return true;
+}
+
+
+std::vector<geometry_msgs::PoseArray> toPosesMsgs(const std::vector<tool_path_planner::ProcessPath>& paths)
+{
+  std::vector<geometry_msgs::PoseArray> poseArrayVector;
+  for(int j = 0; j < paths.size(); ++j)
+  {
+     geometry_msgs::PoseArray poses;
+     poses.header.seq = j;
+     poses.header.stamp = ros::Time::now();
+     poses.header.frame_id = "0";
+     for(int k = 0; k < paths[j].line->GetPoints()->GetNumberOfPoints(); ++k)
+     {
+        geometry_msgs::Pose pose;
+        double pt[3];
+
+        // Get the point location
+        paths[j].line->GetPoints()->GetPoint(k, pt);
+        pose.position.x = pt[0];
+        pose.position.y = pt[1];
+        pose.position.z = pt[2];
+
+        // Get the point normal and derivative for creating the 3x3 transform
+        double* norm =
+            paths[j].line->GetPointData()->GetNormals()->GetTuple(k);
+        double* der =
+            paths[j].derivatives->GetPointData()->GetNormals()->GetTuple(k);
+
+        // perform cross product to get the third axis direction
+        Eigen::Vector3d u(norm[0], norm[1], norm[2]);
+        Eigen::Vector3d v(der[0], der[1], der[2]);
+        Eigen::Vector3d w = u.cross(v);
+        w.normalize();
+
+        // after first cross product, u and w will be orthogonal.
+        // Perform cross product one more time to make sure that v is perfectly
+        // orthogonal to u and w
+        v = u.cross(w);
+        v.normalize();
+
+        Eigen::Affine3d epose = Eigen::Affine3d::Identity();
+        epose.matrix().col(0).head<3>() = v;
+        epose.matrix().col(1).head<3>() = -w;
+        epose.matrix().col(2).head<3>() = u;
+        epose.matrix().col(3).head<3>() = Eigen::Vector3d(pt[0], pt[1], pt[2]);
+
+        tf::poseEigenToMsg(epose, pose);
+
+        // push back new matrix (pose and orientation), this makes one long
+        // vector may need to break this up more
+        poses.poses.push_back(pose);
+
+      }
+      poseArrayVector.push_back(poses);
+    }
+
+  return poseArrayVector;
+}
+
+tool_path_planner::ProcessTool fromTppMsg(const noether_msgs::ToolPathConfig& tpp_msg_config)
+{
+  return tool_path_planner::ProcessTool{.pt_spacing = tpp_msg_config.pt_spacing,
+    .line_spacing = tpp_msg_config.line_spacing,
+    .tool_offset = tpp_msg_config.tool_offset,
+    .intersecting_plane_height = tpp_msg_config.intersecting_plane_height,
+    .min_hole_size = tpp_msg_config.min_hole_size,
+    .min_segment_size = tpp_msg_config.min_segment_size
+  };
+}
+
+noether_msgs::ToolPathConfig toTppMsg(const tool_path_planner::ProcessTool& tool_config)
+{
+  noether_msgs::ToolPathConfig tpp_config_msg;
```

Comment:
This syntax is more usual. Why the change from function to function? I'd rather we use this syntax, but we should be consistent in our syntax.

Reply:
Since this `noether_msgs::ToolPathConfig` is a ros message generated struct it would not allow me to use that other initialization syntax therefore I had to go with the more standard approach.